statsmodels/tsa/automatic/transform.py

```diff
@@ -0,0 +1,51 @@
+from scipy.stats.mstats import gmean
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+from statsmodels import datasets
+
+
+def create_exog(nobs, d=0, seasonal_periods=1):
+    t = np.arange(1, nobs+1)
+    exog = []
+    for i in range(d+1):
+        exog.append(t**i)
+    if seasonal_periods > 1:
+        season = np.zeros((seasonal_periods-1, nobs))
+        for s in range(0, seasonal_periods):
+            for i in range(1, seasonal_periods):
+                for j in range(nobs):
+                    if j % seasonal_periods == i-1:
+                        season[i-1][j] = 1
+        for series in season:
+            exog.append(series)
+    return exog
+
+
+def predict_lambda(endog, lambda_range, nobs, d, seasonal_periods, offset=True):
+
+    for val in endog:
```

Comment:
A loop is probably not as efficient as using a numpy function, so you might want something like:

```python
if np.any(endog < 0):
    raise ValueError('...')
```

Reply:
Yeah, you're right. I'll make the changes.